### PR TITLE
Max touched orders

### PR DESCRIPTION
**Pull request by bh2smith**
In order to prevent an unfair advantage to first solution submitter, we must restrict the number of touched orders to something that only uses no more than 4 Million gas (i.e. approx 25 orders). This is based on the gas estimations made in #231

### CODE REVIEW

**Review by josojo — APPROVED** [review]
Great! 

A hard limit makes it much easier to ensure that there is fair bidding between different parties sending their transactions under slightly different circumstances.

Though, we might want to re-evaluate the number 25, once the smart contract is finished by running all the different gas estimates again.

File: contracts/StablecoinConverter.sol [hunk 1]
```diff
@@ -179,6 +179,7 @@ contract StablecoinConverter is EpochTokenLocker {
         );
         require(tokenIdsForPrice[0] == 0, "fee token price has to be specified");
         require(checkPriceOrdering(tokenIdsForPrice), "prices are not ordered by tokenId");
+        require(owners.length <= 25, "Max number of touched orders is 25");
```
Comment (fleupold) [inline contracts/StablecoinConverter.sol]: Maybe make 25 a constant?

**Review by fleupold — APPROVED** [review]
<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia1.giphy.com%2Fmedia%2FfItgT774J3nWw%2Fgiphy-downsized-medium.gif&token=count" data-gitmeme-token="count"></a>

File: contracts/StablecoinConverter.sol (hunk 1 above)
Comment (bh2smith) [reply contracts/StablecoinConverter.sol]: Check!

**Pull request merged**